From e0f7ea536082a39c2b1ffb73b226ff1bdefc23d9 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Tue, 12 Jun 2018 21:17:33 +0300
Subject: [PATCH 4/5] LineageParts: Complete and correct SMS limits port

* Add missing option entries from previous versions
* Correct string, the SMS limit is for 1 minute since jb 4.2
* Use improved strings from previous versions

Change-Id: I78e9fe9c2f0cb9c934607300221be0335c904d49
---
 res/values/arrays.xml                                     | 4 ++++
 res/values/strings.xml                                    | 4 +++-
 .../lineageos/lineageparts/trust/TrustPreferences.java    | 8 ++++++++
 3 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/res/values/arrays.xml b/res/values/arrays.xml
index 765c109..5cb2586 100644
--- a/res/values/arrays.xml
+++ b/res/values/arrays.xml
@@ -423,18 +423,22 @@
 
     <!-- Sms limit -->
     <string-array name="sms_security_check_limit_entries" translatable="false">
+        <item>@string/sms_security_check_limit_always_confirm</item>
         <item>5</item>
         <item>10</item>
         <item>30</item>
         <item>50</item>
         <item>100</item>
+        <item>200</item>
     </string-array>
 
     <string-array name="sms_security_check_limit_values" translatable="false">
+        <item>0</item>
         <item>5</item>
         <item>10</item>
         <item>30</item>
         <item>50</item>
         <item>100</item>
+        <item>200</item>
     </string-array>
 </resources>
diff --git a/res/values/strings.xml b/res/values/strings.xml
index 1ae9c6b..613678b 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -731,5 +731,7 @@
 
     <!-- Sms limit -->
     <string name="sms_security_check_limit_title">SMS message limit</string>
-    <string name="sms_security_check_limit_summary">Display an alert when more than %s SMS messages are being sent in 15 minutes</string>
+    <string name="sms_security_check_limit_summary">Apps can send %d messages in 1 minute before requiring confirmation</string>
+    <string name="sms_security_check_limit_summary_none">Apps are not allowed to send any messages without confirmation</string>
+    <string name="sms_security_check_limit_always_confirm">Always confirm</string>
 </resources>
diff --git a/src/org/lineageos/lineageparts/trust/TrustPreferences.java b/src/org/lineageos/lineageparts/trust/TrustPreferences.java
index 98b4310..aba31f1 100644
--- a/src/org/lineageos/lineageparts/trust/TrustPreferences.java
+++ b/src/org/lineageos/lineageparts/trust/TrustPreferences.java
@@ -189,9 +189,17 @@ public class TrustPreferences extends SettingsPreferenceFragment {
         return true;
     }
 
+    private void updateSmsSecuritySummary(int selection) {
+        String message = selection > 0
+                ? getContext().getString(R.string.sms_security_check_limit_summary, selection)
+                : getContext().getString(R.string.sms_security_check_limit_summary_none);
+        mSmsLimitPref.setSummary(message);
+    }
+
     private boolean onSmsLimitChanged(Integer value) {
         Settings.Global.putInt(getContext().getContentResolver(),
                 Settings.Global.SMS_OUTGOING_CHECK_MAX_COUNT, value);
+        updateSmsSecuritySummary(value);
         return true;
     }
 
-- 
2.17.1

